plugin/SearchBarItem.qml

```diff
@@ -0,0 +1,248 @@
+**     * Redistributions in binary form must reproduce the above copyright
+**       notice, this list of conditions and the following disclaimer in the
+**       documentation and/or other materials provided with the distribution.
+**     * Neither the name of the Jolla Ltd nor the
+**       names of its contributors may be used to endorse or promote products
+**       derived from this software without specific prior written permission.
+**
+** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+** ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+** WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE LIABLE FOR
+** ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+** (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+** LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+** ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+** (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+** SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**
+****************************************************************************************/
+
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+
+BackgroundItem {
+    id: root
+
+    property bool iconized: true
+    property int modelCount: -1
+    property real iconizedWidth
+
+    property real _margin: Math.max((iconizedWidth - searchIcon.width) / 2., 0.)
+
+    signal requestSearch(string text)
+    signal requestPreviousMatch()
+    signal requestNextMatch()
+    signal requestCancel()
+
+    onClicked: iconized = false
+    onIconizedChanged: if (!iconized) searchField.forceActiveFocus()
+
+    states: [State {
+                 name: "extended"
+                 when: !root.iconized
+                 PropertyChanges {
+                     target: root
+                     _margin: Theme.horizontalPageMargin
+                 }
+             },
+             State {
+                 name: "iconized"
+                 when: root.iconized
+                 PropertyChanges {
+                     target: root
+                     width: iconizedWidth
+                 }
+             }]
+    transitions: Transition {
+        NumberAnimation {
+            properties: "_margin, width"
+            easing.type: Easing.InOutQuad
+            duration: 400
+        }
+    }
+    highlighted: down || searchIcon.down
+
+    IconButton {
+        id: searchIcon
+        anchors {
+            left: parent.left
+            leftMargin: _margin
+        }
+        width: icon.width
+        height: parent.height
+        icon.source: "image://theme/icon-m-search"
+        highlighted: down || root.down || searchField.activeFocus
+
+        enabled: root.enabled
```

Comment:
Enabled is a special property that has by default inherited value from parent, so these lines shouldn't have any effect. Also enabled is not set false anywhere?